master/internal/job/job.go

```diff
@@ -124,134 +137,117 @@ func (j *Jobs) getJobs(
-					errors = append(errors, err.Error())
-				}
-			case *jobv1.QueueControl_Weight:
-				if action.Weight <= 0 {
-					errors = append(errors, "weight must be greater than 0")
-					continue
-				}
-				resp := ctx.Ask(jobActor, sproto.SetGroupWeight{
-					Weight: float64(action.Weight),
-				})
-				if err := resp.Error(); err != nil {
-					errors = append(errors, err.Error())
-				}
-			case *jobv1.QueueControl_ResourcePool:
-				if action.ResourcePool == "" {
-					errors = append(errors, "resource pool must be set")
-					continue
-				}
-				resp := ctx.Ask(jobActor, sproto.SetResourcePool{
-					ResourcePool: action.ResourcePool,
-				})
-				if err := resp.Error(); err != nil {
-					errors = append(errors, err.Error())
-				}
-			case *jobv1.QueueControl_AheadOf:
-				if err := j.rm.MoveJob(ctx, sproto.MoveJob{
-					ID:     jobID,
-					Anchor: model.JobID(action.AheadOf),
-					Ahead:  true,
-				}); err != nil {
-					errors = append(errors, err.Error())
-				}
-			case *jobv1.QueueControl_BehindOf:
-				if err := j.rm.MoveJob(ctx, sproto.MoveJob{
-					ID:     jobID,
-					Anchor: model.JobID(action.BehindOf),
-					Ahead:  false,
-				}); err != nil {
-					errors = append(errors, err.Error())
-				}
-			default:
-				ctx.Respond(fmt.Errorf("unexpected action: %v", action))
-				return nil
+		case *jobv1.QueueControl_Weight:
+			if action.Weight <= 0 {
+				errors = append(errors, "weight must be greater than 0")
+				continue
 			}
+			resp := j.system.Ask(jobActor, sproto.SetGroupWeight{
+				Weight: float64(action.Weight),
+			})
+			if err := resp.Error(); err != nil {
+				errors = append(errors, err.Error())
+			}
+		case *jobv1.QueueControl_ResourcePool:
+			if action.ResourcePool == "" {
+				errors = append(errors, "resource pool must be set")
+				continue
+			}
+			resp := j.system.Ask(jobActor, sproto.SetResourcePool{
+				ResourcePool: action.ResourcePool,
+			})
+			if err := resp.Error(); err != nil {
+				errors = append(errors, err.Error())
+			}
+		case *jobv1.QueueControl_AheadOf:
+			if err := j.rm.MoveJob(j.system, sproto.MoveJob{
+				ID:     jobID,
+				Anchor: model.JobID(action.AheadOf),
+				Ahead:  true,
+			}); err != nil {
+				errors = append(errors, err.Error())
+			}
+		case *jobv1.QueueControl_BehindOf:
+			if err := j.rm.MoveJob(j.system, sproto.MoveJob{
+				ID:     jobID,
+				Anchor: model.JobID(action.BehindOf),
+				Ahead:  false,
+			}); err != nil {
+				errors = append(errors, err.Error())
+			}
+		default:
+			return fmt.Errorf("unexpected action: %v", action)
 		}
-		if len(errors) == 1 {
-			ctx.Respond(fmt.Errorf(errors[0]))
-		} else if len(errors) > 1 {
-			ctx.Respond(fmt.Errorf("encountered the following errors: %s", strings.Join(errors, ", ")))
-		}
-	default:
-		return actor.ErrUnexpectedMessage(ctx)
+	}
+	if len(errors) == 1 {
+		return fmt.Errorf(errors[0])
+	} else if len(errors) > 1 {
+		return fmt.Errorf("encountered the following errors: %s", strings.Join(errors, ", "))
```

Comment:
hashicorp multierror? :shrug:

Reply:
yeah, I think the error handling could be improved here. Continuing on job not found and using `errors.Join` (or a type that implements `Unwrap() []error`) might be preferable. My main concern was preserving current behavior but can update as another PR.